src/components/pages/__tests__/dashboard.test.tsx

```diff
@@ -1,10 +1,37 @@
-import { render, screen } from '@testing-library/react'
+import { configureStore } from '@reduxjs/toolkit'
+import { render, renderHook, screen, waitFor } from '@testing-library/react'
+import { Provider } from 'react-redux'
+
+import { useGetProductsByVendorQuery } from '@/lib/redux/features/product/api.ts'
+import { api } from '@/lib/redux/services/api.ts'
+import { withWrappers } from '@/lib/testing/utils.tsx'
 
 import DashboardPage from '../dashboard.tsx'
 
 describe('<DashboardPage />', () => {
+  it('should handle null response in providesTags', async () => {
+    const store = configureStore({
+      reducer: {
+        [api.reducerPath]: api.reducer,
+      },
+      middleware: getDefaultMiddleware =>
+        getDefaultMiddleware().concat(api.middleware),
+    })
+
+    const wrapper = ({ children }: { children: React.ReactNode }) => (
+      <Provider store={store}>{children}</Provider>
+    )
+
+    const { result } = renderHook(() =>
+      useGetProductsByVendorQuery({ id: '-1' }), { wrapper },
+    )
+
+    await waitFor(() => expect(result.current.isFetching).toBe(false))
+    expect(result.current.data).toBeNull()
+  })
+
```

Comment:
Instead of using this pattern, let's use MSW's `.use()`

First of all, let's create the server instance in `@/lib/msw/index.ts`

```ts
// @/lib/msw/index.ts [NOTE: this is a new file]
import { setupServer } from 'msw/node'

export const mswServer = setupServers(...handlers) // import handlers here
```

In `vitest.setup.ts`, change things to
```ts
import './src/index.css'
import '@testing-library/jest-dom/vitest'

import { cleanup } from '@testing-library/react'
import { afterEach } from 'vitest'

import { mswServer } from '@/lib/msw/index.ts'

// Start server before all tests
beforeAll(() => mswServer.listen({ onUnhandledRequest: 'error' }))

// Close server after all tests
afterAll(() => mswServer.close())

// Reset handlers after each test `important for test isolation`
afterEach(() => mswServer.resetHandlers())

// Runs a clean after each test case (e.g. clearing jsdom)
afterEach(() => {
  cleanup()
})
```

And in this test file, we can import `mswServer` like so:
```ts
import { mswServer } from '@/lib/msw/index.ts'

describe('SomeTest', () => {
  it('should ...', () => {
    mswServer.use(errorHandler)
  })
})
```

For `.use()` reference, check [this](https://mswjs.io/docs/api/setup-server/use/) out
Any handlers that is prepended by `.use` will be reset after each test by [`mswServer.resetHandlers()`](https://mswjs.io/docs/api/setup-server/reset-handlers) this is already done in `vitest.setup.ts` so don't worry about that